Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationFailureRepository.cs

```diff
@@ -0,0 +1,234 @@
+
+      var failureContainer = GetOrCreateRowFailureContainer(rowObject);
+      failureContainer.AddCellFailures(columnDefinition, validationFailures);
+    }
+
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailuresForBocList (bool markAsHandled)
+    {
+      var failures = GetUnhandledValidationFailuresForBocListInternal();
+
+      if (markAsHandled)
+        failures = failures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return failures.Select(f => f.Failure).ToList();
+    }
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetValidationUnhandledFailuresForBocListAndContainingDataRowsAndDataCells (bool markAsHandled)
+    {
+      var failures = GetUnhandledValidationFailuresForBocListInternal();
+
+      foreach (var failureContainer in _rowValidationFailures.Values)
+      {
+        failures = failures.Concat(failureContainer.GetValidationFailuresForDataRow().Where(f => !f.Handled));
+        failures = failures.Concat(failureContainer.GetAllCellFailuresForRow().Where(f => !f.Handled));
+      }
+
+      if (markAsHandled)
+        failures = failures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return failures.Select(f => f.Failure).ToList();
+    }
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailuresForDataRow (IBusinessObject rowObject, bool markAsHandled)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rowObject), rowObject);
+
+      var failureContainer = _rowValidationFailures.GetValueOrDefault(rowObject);
+      if (failureContainer == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = failureContainer.GetValidationFailuresForDataRow().Where(f => !f.Handled);
+
+      if (markAsHandled)
+        failures = failures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return failures.Select(f => f.Failure).ToList();
+    }
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailuresForDataRowAndContainingDataCells (IBusinessObject rowObject, bool markAsHandled)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rowObject), rowObject);
+
+      var failureContainer = _rowValidationFailures.GetValueOrDefault(rowObject);
+      if (failureContainer == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = failureContainer.GetValidationFailuresForDataRow().Where(f => !f.Handled);
+      failures = failures.Concat(failureContainer.GetAllCellFailuresForRow().Where(f => !f.Handled));
+
+      if (markAsHandled)
+        failures = failures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return failures.Select(f => f.Failure).ToList();
+
+    }
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailuresForDataCell (IBusinessObject rowObject, IBocColumnDefinitionWithValidationSupport columnDefinition, bool markAsHandled)
```

Comment:
The line is to long as far as I can tell without opening the code in the IDE. please verify and add linewraps. Please also check the rest of the files.